dashboard/app/auth.go

```diff
@@ -116,34 +114,34 @@ func (auth *authEndpoint) determineAuthSubj(authHeader []string) (string, error)
 		return "", err
 	}
 	if claims.Audience != dashapi.DashboardAudience {
-		err := fmt.Errorf("Unexpected audience %v %v", claims.Audience, claims)
+		err := fmt.Errorf("unexpected audience %v %v", claims.Audience, claims)
 		return "", err
 	}
 	if claims.Expiration.Before(time.Now()) {
-		err := fmt.Errorf("Token past expiration %v", claims.Expiration)
+		err := fmt.Errorf("token past expiration %v", claims.Expiration)
 		return "", err
 	}
 	return oauthMagic + claims.Subject, nil
 }
 
 // Verifies that the given credentials are acceptable and returns the
 // corresponding namespace.
-func checkClient(name0, secretPassword, oauthSubject string) (string, error) {
+func checkClient(conf *GlobalConfig, name0, secretPassword, oauthSubject string) (string, error) {
```

Comment:
We currently never pass config around and rely on the global one.
The tests use the config defined in app_test.go. So I would add there a namespace for auth testing with a set of clients with different passwords/oauth.